lib/Sema/ConstraintSystem.h

```diff
@@ -3024,6 +3024,11 @@ class ConstraintSystem {
   void resolveOverload(ConstraintLocator *locator, Type boundType,
                        OverloadChoice choice, DeclContext *useDC);
 
+  /// Simplifies a type by replacing type variables with the result of
+  /// \c getFixedTypeFn and performing lookup on dependent member types.
+  Type simplifyTypeImpl(Type type,
```

Comment:
This method should probably be `private:`

Reply:
`Solution` calls this function. Should `Solution` be a friend of `ConstraintSystem`?

Reply:
That would make sense.